front50-azure/src/main/java/com/netflix/spinnaker/front50/model/AzureStorageService.java

```diff
@@ -58,11 +59,14 @@ private CloudBlobContainer getBlobContainer() {
       try {
         blobContainer = getBlobClient().getContainerReference(this.containerName);
         blobContainer.createIfNotExists();
-        BlobContainerPermissions permissions = new BlobContainerPermissions();
+        // removing the public access enforcement, so that we can leverage private and public AZ
+        // https://github.com/spinnaker/spinnaker/issues/6218
+        /*BlobContainerPermissions permissions = new BlobContainerPermissions()
         permissions.setPublicAccess(BlobContainerPublicAccessType.CONTAINER);
-        blobContainer.uploadPermissions(permissions);
+        blobContainer.uploadPermissions(permissions);*/
       } catch (Exception e) {
-        // log exception
+        // log exception is missed
+        log.error(e.getMessage() + e.getCause() + e.getStackTrace());
```

Comment:
Hey,
Does creating a container default to making it private?  It might still be a good idea to explicitly set it to private when created anyway.  So there is no ambiguity.

I'm not sure if Spinnaker would end up locking itself out of accessing the blob storage it creates though.  I guess it is up to the creator of the principal to set it correctly.

Also, a suggestion on logging the error so that it is more in line with how the other exceptions are logged.

```java
     try {
        blobContainer = getBlobClient().getContainerReference(this.containerName);
        if (blobContainer.createIfNotExists()) {
          BlobContainerPermissions permissions = new BlobContainerPermissions();
          permissions.setPublicAccess(BlobContainerPublicAccessType.OFF);
          blobContainer.uploadPermissions(permissions);
        }
      } catch (Exception e) {
        log.error(
          "Exception occurred getting/creating the blob container: {} ",
          value("exception", e.getMessage()));
        blobContainer = null;
      }
```

Reply:
I think this is a better implementation off hand - but it might be worth allowing the ability to customize these permissions.

Reply:
@jasonmcintosh ,
Which part do you  think should be configurable?  I'm thinking that only the access upon creation should configurable.  Once set it should likely never touch the permissions again else it might step on something external managing the permissions.  In our case, the Storage Container is created and maintained by Terraform.  Last thing we would expect is an application changing it out from under us.

Also, do you know where the configuration is maintained in code so I can take a look?

Thanks!

Reply:
OK read through some of this a bit more.  I think yeah changing the creation to set an explicit anonymous access deny is LIKELY a good thing.  WIth that said, I'm not as familiar with azure permissions stuff - so can't entirely say what should/shouldn't be.  I like the code change you've got up above a bit more personally :)